doc/overview.md

```diff
@@ -0,0 +1,129 @@
+# Twyla Service
+
+The Twyla Service library provides primitives to build micro services that just
+work within the Twyla platform.
+
+## RPC
+
+Remote procedure calls are used for synchronous communication between services
+and to communicate with the integrations. RPC is done via the HTTP protocol.
+
+The RPC implementation is a wrapper around the Python `aiohttp` library.
+
+### Using RPC requests
+
+    import twyla.service.rpc as rpc
+
+    res = rpc.call('navitaire-api.get_booking', data={'pnr': 'A1324B'})
+
+
+### Managing Changes
+
+See the Managing Changes section of the Event Bus. Most of those points apply to
+RPC as well.
+
+
+## Event Bus
+
+Events are used for asynchronous communication between services.
+
+Events are emitted by services to either signal a state change (e.g. change of
+user data, start of a proxy session to a 3rd party bot or service) or request
+information from other services within the platform (e.g. query information
+about a booking).
+
+The letter is similar to RPC but decouples the request from awaiting and
+processing the response. The main disadvantage of this method of requesting
+information compared to RPC is the lack of a guaranteed response (RPC requests
+will always have a success, failure, or timeout response).
+
+
+### Creating Events
+
+Events have to implement a specific contract that defines the structure and
+value types within the structure of the event.
+
+The event can either either implement one of the predefined contracts in this
+library or an automatically discovered contract that gets exposed by the
+services that listen to a particular event.
+
+    import twyla.service.event as event
+
+    generic = event.contracts.Generic(name='generic_event', data={})
+    generic.emit(data={'id': 'some-event-id'})
+
+    create_user = event.discover('create_user')
```

Comment:
The discovery idea for schemata is still a bit rough, but I think this will really help to store the definitions somewhere outside of the service scope.. even though that probably creates other problems 🤔

Reply:
Maybe `load()` is a better name?

Reply:
Combining pydantic and schemata of some other format will be a bit of work as well.